master/internal/scheduler/filterable_view.go

```diff
@@ -25,14 +25,16 @@ func newProvisionerView(provisionerSlotsPerInstance int) *FilterableView {
 	}
 }
 
-func schedulableTaskFilter(provisionerSlotsPerInstance int) func(*Task) bool {
+func schedulableTaskFilter(
+	provisionerSlotsPerInstance int,
+) func(*AllocateRequest, *ResourcesAllocated) bool {
 	// We only tell the provisioner about pending tasks that are compatible with the
 	// provisioner's configured instance type.
-	return func(task *Task) bool {
-		slotsNeeded := task.SlotsNeeded()
+	return func(req *AllocateRequest, assigned *ResourcesAllocated) bool {
+		slotsNeeded := req.SlotsNeeded
 
 		switch {
-		case task.state != taskPending:
+		case assigned != nil && len(assigned.Allocations) > 0:
```

Comment:
non-bcloking: maybe add a helper function to task_list for `isAssigned()`

Reply:
I found this is the only place that needs to determine if a task is allocated. I will rewrite this part anyways when I refactor the protocol between the scheduler and the provisioner. So won't bother doing that now.